src/main/java/com/microsoft/graph/models/extensions/Multipart.java

```diff
@@ -19,6 +19,10 @@
 public class Multipart {
 	private String boundary;
 	private static final String RETURN = "\r\n";
+	/**
+     * The encoding type for getBytes
+     */
+    static final String JSON_ENCODING = "UTF-8";
```

Comment:
The content is not necessarily JSON! You need to investigate what the encoding is for the Multipart content. I'm assuming this is for SMTP Multipart data.  Please justify your choice of encoding (and put it in a comment in the source).

Reply:
I have updated String names. Also, https://developer.microsoft.com/en-us/graph/docs/concepts/onenote_images_files#add-a-file-attachment mentions "send the binary data, don't use Base64 or otherwise encode it" and https://en.wikipedia.org/wiki/MIME#Multipart_messages explains to use encoding 7bit, 8bit or binary like UTF-8.

Reply:
The job does not seem to be done yet. Can the user set the Content-Transfer-Encoding MIME header (which you would then have to honour)?

Reply:
@davidmoten Currently we don't appear to support setting any headers for parts other than Content-Type and Content-Transfer-Encoding.  If we did then the user would be highly recommended to pass the properly encoded bytes in as a byte array.

@NakulSabharwal For the overload that accepts a name and creates a Content-Disposition header, we can't really just use UTF-8 decoding on that' string.  MIME headers in multipart only support US-ASCII and percent encoding.  The full spec for multipart/form-data is here https://tools.ietf.org/html/rfc7578 and it was updated fairly recently.

We should get together and consider what multipart features we need to support for existing Graph APIs and write up requirements across all the languages.  Assuming UTF-8 strings for JSON is fine, but it's not for HTML.  We also need to look into whether we need to support filename as a content disposition parameter.

Handling multipart properly is tricky, and I want to make sure we are doing this consistently across all the languages and that we support all the features we need for Graph.

Unfortunately I'm out this week, but maybe we can gather some requirements and meet next week to nail down exactly what features we are trying to support, that would be great.  /cc @deepak2016